torch/csrc/api/src/optim/adam.cpp

```diff
@@ -31,24 +31,24 @@ void Adam::step() {
     auto& exp_average_sq = buffer_at(exp_average_sq_buffers, i);
 
     buffer_at(step_buffers, i) += 1;
+    const auto bias_correction1 =
+        1 - std::pow(options.beta1_, buffer_at(step_buffers, i));
+    const auto bias_correction2 =
+        1 - std::pow(options.beta2_, buffer_at(step_buffers, i));
 
     exp_average.mul_(options.beta1_).add_(p.grad(), 1 - options.beta1_);
     exp_average_sq.mul_(options.beta2_)
         .addcmul_(p.grad(), p.grad(), 1 - options.beta2_);
 
-    Tensor denom = exp_average_sq;
+    Tensor denom = exp_average_sq / bias_correction2;
```

Comment:
I believe `bias_correction2`  should have a square root, isn't that so?

Reply:
Sorry, I forgot to remove the `sqrt` in this line. I fixed it in 72c6582.
We are doing the `denom.sqrt()` at the end of `adam.cpp` whereas in python scripts the square root is calculated before `addcdiv_ `